src/graphics/shadow.odin

```diff
@@ -0,0 +1,351 @@
+package graphics
+
+import la "core:math/linalg"
+import vk "vendor:vulkan"
+import "pkgs:vma"
+
+import "builders"
+
+Shadow :: struct{
+    view: mat4,
+    proj: mat4,
+}
+
+ShadowContext :: struct{
+    // Pipeline
+    mat: ^Material,
+
+    // Uniforms
+    data: []Shadow,
+    buffer: Buffer,
+
+    // Rendering
+    images: []Image,
+    framebuffers: []vk.Framebuffer,
+    render_pass: vk.RenderPass,
+    
+    // Constants
+    frame_count: int,
+    extent: vk.Extent2D,
+}
+
+shadow_create_image :: proc(device: vk.Device, extent: vk.Extent2D) -> (image: Image) {
```

Comment:
i'm on my phone but is there a way to refactor buffers module so that it also can make shadow images/views? Id like to keep having buffer creation in one place